sys/compat/freebsd64/freebsd64_misc.c

```diff
@@ -691,19 +697,19 @@ freebsd64_copyout_strings(struct image_params *imgp, uintcap_t *stack_base)
 	 * Allocate room for the argument and environment strings.
 	 */
 	destp -= ARG_MAX - imgp->args->stringspace;
-	destp = __builtin_align_down(destp, sizeof(uint64_t));
-	ustringp = destp;
+	destp = rounddown2(destp, sizeof(uint64_t));
+	ustringp = cheri_setbounds(destp, ARG_MAX - imgp->args->stringspace);
```

Comment:
Is this missing in the purecap kernel as well?

Reply:
I haven't looked, but I'll merge dev to the purecap branch again "soon"